src/lib/pci_virtio_net_vpnkit.c

```diff
@@ -423,7 +493,7 @@ vpnkit_create(struct pci_vtnet_softc *sc, const char *opts)
 			goto err;
 		}
 
-		if (vpnkit_connect(fd, uuid_string, &state->vif) == 0)
+		if (vpnkit_connect(fd, uuid_string, &state->vif, (ipv4 == NULL ? NULL : &preferred_ipv4.s_addr)) == 0)
```

Comment:
Perhaps you could arrange for `s_addr` to be zero being the default/unconfigured action, which would avoid the conditionals and pointer argument here, i.e. vpnkit_connect could just take a plain `in_addr_t`.